### PR TITLE
Add helm addon charts

### PR DESCRIPTION
**Pull request by SpringUpenn**
Since addon templates in https://github.com/istio/istio/tree/master/install/kubernetes/templates/addons are deprecating, and it will use helm template instead, this PR is for deprecating old templates and use helm instead. 
The helm addon charts contains union of existing helm addon templates and old-way addon templates.

### CODE REVIEW

**Comment by ostromart** [comment]
Why are these changes required for istio on gke but not regular istio?

**Comment by SpringUpenn** [comment]
It is needed for both istio on gke and regular istio. 
Regular istio has the plan to deprecate old-way addon templates in https://github.com/istio/istio/tree/master/install/kubernetes/templates/addons and use helm instead, I am helping on this. 
Once it is merged, it can benefit istio on gke as well, so istio on gke yaml files can all be generated with helm, without mix of helm and updateVersion.sh

**Comment by mandarjog** [comment]
Thanks Emily for the PR. This is super helpful. Will evaluate if this should go into 0.8.

/ok-to-test

File: install/kubernetes/helm/istio/charts/zipkin-to-stackdriver/templates/deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+      name: zipkin-to-stackdriver
+      labels:
+        app: zipkin-to-stackdriver
+      annotations:
+        sidecar.istio.io/inject: "false"
+    spec:
+      containers:
+        - name: zipkin-to-stackdriver
+          image: "{{ .Values.image.repository }}"
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
```
[the remark this answers is not in the document]
Comment (sdake) [reply install/kubernetes/helm/istio/charts/zipkin-to-stackdriver/templates/deployment.yaml]: ya - we should really remove each service's pull policy from the values.yaml and add a comment that we have a unified pull policy for new contribs.

File: install/kubernetes/helm/istio/values-istio-all.yaml [hunk 2]
```diff
@@ -22,3 +22,6 @@ servicegraph:
 
 prometheus:
   enabled: true
+
+zipkin-to-stackdriver:
+  enabled: true
```
Comment (sdake) [inline install/kubernetes/helm/istio/values-istio-all.yaml]: do we want this enabled by default? (I don't know enough about the addons to intelligently answer the question myself)

File: install/kubernetes/helm/istio/values.yaml [hunk 3]
```diff
@@ -423,3 +423,17 @@ zipkin:
...
+zipkin-to-stackdriver:
+  enabled: false
+  replicaCount: 1
+  service:
+    name: zipkin
+    externalPort: 9411
+    internalPort: 9411
+  image:
+    repository: gcr.io/stackdriver-trace-docker/zipkin-collector
+    pullPolicy: IfNotPresent
```
[the remark this answers is not in the document]
Comment (sdake) [reply install/kubernetes/helm/istio/values.yaml]: can probably just delete this pull policy (line 439).

File: install/kubernetes/helm/istio/values-istio-all.yaml (hunk 2 above)
Comment (SpringUpenn) [reply install/kubernetes/helm/istio/values-istio-all.yaml]: by default it is false, it is set in https://github.com/SpringUpenn/istio/commit/b4b25f92f9a3140d60b1ce65f4a63e4491e4c820#diff-7dad29cba9d2ca3e8570c8f65f4b7e86, this is for values-istio-all.yaml, which wants to enable all addons(based on it sets true for zipkin, grafana etc.), correct me if I understand wrong.

**Pull request closed without merging**